ttnn/cpp/ttnn/device_operation.hpp

```diff
@@ -258,25 +256,28 @@ void launch_on_worker_thread(auto cq_id, auto device_operation_id, const auto& o
         device_operation_t::validate_on_program_cache_miss(operation_attributes, tensor_args);
     }
 
-    const auto enqueue_or_launch_program = [=](Program& program) {
+    const auto enqueue_or_launch_program = [=](ProgramHandle program) {
         if (USE_FAST_DISPATCH) {
             ZoneScopedN("EnqueueProgram");
             auto& queue = device->command_queue(cq_id);
             tt::tt_metal::EnqueueProgram(queue, program, false);
         } else {
             ZoneScopedN("LaunchProgram");
-            tt::tt_metal::detail::LaunchProgram(device, program);
+            auto* program_ptr = tt::tt_metal::ProgramPool::instance().get_program(program);
+            tt::tt_metal::detail::LaunchProgram(device, *program_ptr);
```

Comment:
(I know the PR is still a draft, just curious)

is this temporary and won't be needed once LaunchProgram supports ProgramHandle?